plugins/Widgetize/vue/src/ExportWidget/ExportWidget.vue

```diff
@@ -0,0 +1,169 @@
+          language=fr. For example, for idSite=1 and date=yesterday, you can write:
+          <br />
+        </p>
+        <pre
+          v-text="dashboardCode"
+          v-select-on-focus="{}"
+        ></pre>
+        <p>
+          <br />
+          You can also widgetize the all websites dashboard in an IFRAME
+          (<a
+            rel="noreferrer noopener"
+            target="_blank"
+            id="linkAllWebsitesDashboardUrl"
+            :href="allWebsitesDashboardUrl"
+          >see example</a>)
+          <br />
+        </p>
+        <pre
+          v-text="allWebsitesDashboardCode"
+          v-select-on-focus="{}"
+        ></pre>
+      </div>
+    </ContentBlock>
+    <ContentBlock content-title="Widgetize reports">
+      <div>
+        <p>Select a report, and copy paste in your page the embed code below the widget:</p>
+        <div>
+          <WidgetPreview />
+        </div>
+        <br class="clearfix" />
+      </div>
+    </ContentBlock>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import {
+  Matomo,
+  ContentIntro,
+  EnrichedHeadline,
+  ContentBlock,
+  SelectOnFocus,
+  MatomoUrl,
+} from 'CoreHome';
+import WidgetPreview from '../WidgetPreview/WidgetPreview.vue';
+
+interface ExportWidgetState {
+  dashboardUrl: string;
+  allWebsitesDashboardUrl: string;
+}
+
+function getIframeCode(iframeUrl: string) {
+  const url = iframeUrl.replace(/"/g, '&quot;');
+  return `<iframe src="${url}" frameborder="0" marginheight="0" marginwidth="0" width="100%" `
+    + 'height="100%"></iframe>';
+}
+
+export default defineComponent({
+  props: {
+    title: {
+      type: String,
+      required: true,
+    },
+  },
+  components: {
+    EnrichedHeadline,
+    ContentBlock,
+    WidgetPreview,
+  },
+  directives: {
+    ContentIntro,
+    SelectOnFocus,
+  },
+  data(): ExportWidgetState {
+    const port = window.location.port === '' ? '' : `:${window.location.port}`;
+    const path = window.location.pathname;
+    const urlPath = `${window.location.protocol}//${window.location.hostname}${port}${path}`;
+
+    return {
+      dashboardUrl: `${urlPath}?${MatomoUrl.stringify({
+        module: 'Widgetize',
+        action: 'iframe',
+        moduleToWidgetize: 'Dashboard',
+        actionToWidgetize: 'index',
+        idSite: Matomo.idSite,
+        period: 'week',
+        date: 'yesterday',
+      })}`,
+      allWebsitesDashboardUrl: `${urlPath}?${MatomoUrl.stringify({
+        module: 'Widgetize',
+        action: 'iframe',
+        moduleToWidgetize: 'MultiSites',
+        actionToWidgetize: 'standalone',
+        idSite: Matomo.idSite,
+        period: 'week',
+        date: 'yesterday',
+      })}`,
+    };
+  },
+  computed: {
+    dashboardCode() {
+      return getIframeCode(this.dashboardUrl);
+    },
+    allWebsitesDashboardCode() {
+      return getIframeCode(this.allWebsitesDashboardUrl);
+    },
+  },
+  methods: {
+    linkTo(params: QueryParameters) {
```

Comment:
Would it maybe make sense to make that method global in some way. Iirc I have seen similar methods in other components already.

Reply:
As far as I know the Vue policy/suggestion is to always use computed properties instead of adding many global functions to Vue apps. This is likely because computed properties are cached, whereas function calls and other code in an component's template will be executed every time a component updates (as far as I know, again). We have two, $sanitize/translate now. translate is used ubiquitously so it would be very inconvenient to make computed properties for everything that calls it. $sanitize isn't and it would probably be faster to use it in computed properties only, really.

I could make these computed properties that just call MatomoUrl given linkTo is usually given hardcoded parameters, but personally I would say its not in the spirit of Vue to add global functions like that, especially when they are only used by a handful of components.

Reply:
Thanks for the explanation. Makes sense now.